at-date-api/src/main/kotlin/com/amerharb/atdate/app.kt

```diff
@@ -0,0 +1,115 @@
+package com.amerharb.atdate
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.engine.embeddedServer
+import io.ktor.server.netty.Netty
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import java.util.*
+
+const val PORT = 8000
+
+fun main() {
+    embeddedServer(Netty, port = PORT, module = Application::module).start(wait = true)
+}
+
+fun Application.module() {
+    routing {
+        get("/") {
+            call.respondText("Hello, world!")
+        }
+
+        post("/encode/{notation}") {
+            val notation = call.parameters["notation"]
+            if (notation == null) {
+                call.respondText("Notation is null")
+                return@post
+            }
+            // if encoding throw error respond with error
+            try {
+                val ad = encode(notation)
+
+                val res = """
+                    hex: ${ad.getHex()}
+                    base64: ${ad.getBase64()}
+            """.trimIndent()
+                call.respondText(res)
+            } catch (e: Exception) {
+                call.respondText(
+                    text = e.message ?: "Error",
+                    contentType = ContentType.Text.Plain,
+                    status = HttpStatusCode.InternalServerError,
+                )
+                return@post
+            }
+        }
+
+        post("/encode/{notation}/hex") {
```

Comment:
Here you're not creating or modifying any resources, you are just retrieving informations based on the input you receive,  therefore using post is not technically wrong, but inappropriate according to restful-api principles 
check this https://restfulapi.net/

Reply:
+1 i think GET make more sence